crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -164,11 +288,16 @@ impl<'db> TypeInferenceBuilder<'db> {
         let function_ty =
             Type::Function(FunctionType::new(self.db, name.id.clone(), decorator_tys));
 
-        let definition = self.index.definition(function);
         self.types.definitions.insert(definition, function_ty);
     }
 
     fn infer_class_definition_statement(&mut self, class: &ast::StmtClassDef) {
+        let definition = self.index.definition(class);
+        let result = infer_definition_types(self.db, definition);
+        self.extend(result);
```

Comment:
This seems to always be the same pattern. Convert node into definition, infer the definition types, extend. 

I think we can create a helper function for this

```
fn infer_definition(&self, node: impl Into<DefinitionNodeKey>) {
        let definition = self.index.definition(node);
        let result = infer_definition_types(self.db, definition);
        self.extend(result);
}
```